lib/appSession.js

```diff
@@ -132,46 +137,44 @@ module.exports = (config) => {
             (i + 1) * CHUNK_BYTE_SIZE
           );
           const chunkCookieName = `${sessionName}.${i}`;
-          res.cookie(chunkCookieName, chunkValue, cookieOptions);
+          cookies.set(chunkCookieName, chunkValue, cookieOptions);
         }
       } else {
-        res.cookie(sessionName, value, cookieOptions);
+        cookies.set(sessionName, value, cookieOptions);
       }
     }
   }
 
-  return (req, res, next) => {
-    if (req.hasOwnProperty(sessionName)) {
+  return (res, sessionStore, cookieStore, cookies, isSecure, next) => {
```

Comment:
res is kept here as its only used to pass to `onHeaders` which expects a Node ServerResponsem which Express' Response extends.